src/provision.ts

```diff
@@ -17,5 +19,8 @@ export const startProvisioning = async () => {
 
   const el = document.createElement("improv-wifi-provision-dialog");
   el.device = device;
+  el.stateUpdateCallback = (state) => {
+    fireEvent(button, "state-changed", state);
```

Comment:
Let's make detail an object so we can expand it while remaining backwards compatibility.
```suggestion
    fireEvent(button, "state-changed", { state });
```

Reply:
It is an object, `stateUpdateCallback` receives an object.